src/jaegertracing/DynamicLoad.cpp

```diff
@@ -41,7 +41,8 @@ static int makeTracerFactory(const char* opentracingVersion,
         return opentracing::incompatible_library_versions_error.value();
     }
 
-    *tracerFactory = new (std::nothrow) jaegertracing::TracerFactory{};
+    const auto jaegerTracerFactory = new (std::nothrow) jaegertracing::TracerFactory(true);
```

Comment:
I am not a fan of this API. Without looking up the definition, what does `ctor(true)` mean?

In the other languages we simply provide a fromEnv() initializer. Not only it is explicit, but it also decouples factory initialization and reading env, e.g. it is very common to first initialize the factory with some default configuration, and then allow it to be overwritten by env. Here it looks like env vars are read immediately.

Reply:
It was originally implemented like that, but @mdouaihy recommended the ctor approach https://github.com/jaegertracing/jaeger-client-cpp/pull/192#issuecomment-570489433

I don’t have an opinion and happily change it to your preference

Reply:
I may be mistaken about the role of TracerFactory. If it's a loading mechanism similar to Java's service loader, then it's probably ok to apply env right in the ctor. While I don't like the bool argument, it's ok for a non-user-facing class.

Reply:
@yurishkuro, In order to create a tracer, a tracer factory is first created and then MakeTracer method is called to create a tracer.

In MakeTracer, a yaml config file sent as argument. Properties of the tracer are read and a Tracer is configured and built

We would like to read also the env variables. Now since this is a behavioral change, we would like to make this behavior activatable.

The problem is that we cannot change the signature of MakeTracer. it's a public method. the only way is activate this behavior via a parameter passed to the factory at construction time.